beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -191,3 +192,86 @@ func ProcessEjections(state *pb.BeaconState) (*pb.BeaconState, error) {
 	}
 	return state, nil
 }
+
+// ProcessValidatorRegistry computes and sets new validator registry fields,
+// reshuffles shard committees and returns the recomputed state.
+//
+// Spec pseudocode definition:
+//	Set state.validator_registry_latest_change_slot = state.slot.
+//	Set state.shard_committees_at_slots[:EPOCH_LENGTH] = state.shard_committees_at_slots[EPOCH_LENGTH:].
+//	Set state.shard_committees_at_slots[EPOCH_LENGTH:] =
+//  	get_new_shuffling(state.latest_randao_mixes[(state.slot - EPOCH_LENGTH) %
+//  	LATEST_RANDAO_MIXES_LENGTH], state.validator_registry, next_start_shard)
+//  	where next_start_shard = (state.shard_committees_at_slots[-1][-1].shard + 1) % SHARD_COUNT.
+func ProcessValidatorRegistry(
+	state *pb.BeaconState) (*pb.BeaconState, error) {
+
+	epochLength := int(params.BeaconConfig().EpochLength)
+	randaoMixesLength := params.BeaconConfig().LatestRandaoMixesLength
+	shardCount := params.BeaconConfig().ShardCount
+	lastShardAndCommitteesAtSlots := len(state.ShardAndCommitteesAtSlots) - 1
+	lastCommiteeFromLastSlot := len(state.ShardAndCommitteesAtSlots[lastShardAndCommitteesAtSlots].ArrayShardAndCommittee) - 1
```

Comment:
Can you shorten these lines? This is very verbose and the line does not fit within the github code review.

If you can make it under 100 chars, that would be nice. I personally try to keep everything under 80 chars, but we don't have a hard limit. Whatever feels right